client/src/unifyfs-sysio.c

```diff
@@ -1609,27 +1609,31 @@ int UNIFYFS_WRAP(lio_listio)(int mode, struct aiocb* const aiocb_list[],
         }
 
         /* update aiocb fields to record error status and return value */
-        ndx = 0;
         for (i = 0; i < reqcnt; i++) {
-            char* buf = reqs[i].buf;
-            for (; ndx < nitems; ndx++) {
+            read_req_t* req = &reqs[i];
+            char* buf = req->buf;
+
+            for (ndx = 0; ndx < nitems; ndx++) {
                 cbp = aiocb_list[ndx];
                 if ((char*)(cbp->aio_buf) == buf) {
```

Comment:
I know you didn't change this line, but I feel like this check may not be a reliable way to match the request.  Is it allowed for two different requests to each provide the same user buffer address?  That wouldn't make a lot of sense, but if it is allowed, then we'll need to find some other way to match the read_request to the aio structure.  Just thought I'd raise this in passing.  If we deem it to be a problem, we can create an issue to fix it later.

Reply:
Oh just thought of a valid case.  This might be a reasonable thing to do if a process writes the same user buffer to two different files or even to two different offsets in the same file.

Reply:
we should capture this defect in an issue and resolve separately.

Reply:
this seems to be resolved with the addition of the aiocb pointer